client/lib/cart-values/index.js

```diff
@@ -106,6 +106,10 @@ function getRefundPolicy( cart ) {
 	return 'genericRefund';
 }
 
+function isPayPalExpressEnabled( cart ) {
+	return cart.allowed_payment_methods.indexOf( 'WPCOM_Billing_PayPal_Express' ) >= 0;
```

Comment:
The previous free trials did a an authorization of credit card, and on that form, PayPal Express was disabled. The current iteration of free trials does not use credit cards, so they will not see this.

And if in future we will again do free trials with card, we can disable PayPal express on backend using the  allowed_payment_methods field :)